pkg/storage/mvcc_incremental_iterator.go

```diff
@@ -630,8 +627,12 @@ func (i *MVCCIncrementalIterator) NextIgnoringTime() {
 
 		// We have encountered an intent but it does not lie in the timestamp span
 		// (startTime, endTime] so we do not throw an error, and attempt to move to
-		// the next valid KV.
+		// the intent's corresponding provisional value.
+		//
+		// TODO(msbulter): investigate if it's clearer for the caller to emit the intent rather than the
+		// provisional value
```

Comment:
I'm not sure it would ever make sense to emit the intent _rather than_ the provisional value, but I could see an argument for emitting both.

Let's also document for posterity that right now some callers _depend_ on seeing the provisional value which is why we emit it.

Reply:
Done.